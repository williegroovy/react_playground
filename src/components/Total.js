import React from 'react';
import PropTypes from "prop-types";
import styled from 'styled-components';
import { formatAsMoney } from '../utils/helpers';
import { StyledFlexAlign } from '../styled-components';

const StyledTotalContainer = styled(StyledFlexAlign)`
  width: 100%;
  display: flex;
  align-items: flex-end;
  height: 30px;
`;

const Total = ({ amount, align }) => {
  return (
    <StyledTotalContainer align={align}>
      Total: ${formatAsMoney(amount)}
    </StyledTotalContainer>
  )
};

Total.propTypes = {
  amount: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired
};

export default Total;