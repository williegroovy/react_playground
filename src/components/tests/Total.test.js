import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Total from '../Total'

describe('<Total />', () => {
  test('should be defined', () => {
    expect(Total).toBeDefined();
  });

  test('should render correctly' , () => {
    const wrapper = shallow(
      <Total amount={100} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render $1.00 when 1 is passed as amount prop' , () => {
    const wrapper = shallow(<Total amount={1} />);
    expect(wrapper.props().children).toEqual(expect.arrayContaining(['Total: $', '1.00']));
  });

  test('should render $10,000.00 when 10000 is passed as amount prop' , () => {
    const wrapper = shallow(<Total amount={10000} />);
    expect(wrapper.props().children).toEqual(expect.arrayContaining(['Total: $', '10,000.00']));
  });

  test('should render $10,000,000.00 when 10000000 is passed as amount prop' , () => {
    const wrapper = shallow(<Total amount={10000000} />);
    expect(wrapper.props().children).toEqual(expect.arrayContaining(['Total: $', '10,000,000.00']));
  });
});