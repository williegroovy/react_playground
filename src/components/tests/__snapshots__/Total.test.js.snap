// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Total
    amount={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "align": undefined,
      "children": Array [
        "Total: $",
        "100.00",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Total: $",
      "100.00",
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
",
          "
  width: 100%;
  display: flex;
  align-items: flex-end;
  height: 30px;
",
        ],
      },
      "displayName": "Styled(styled.div)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "align": undefined,
        "children": Array [
          "Total: $",
          "100.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Total: $",
        "100.00",
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  ",
            [Function],
            "
",
            "
  width: 100%;
  display: flex;
  align-items: flex-end;
  height: 30px;
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
